### ISSUE TITLE
Signal Android APK stuck on Verification screen on a deGoogled GrapheneOS

### ISSUE THREAD

**Issue by ekotysh** [signalapp/Signal-Android#13624]
### Bug description
I am stuck on Verification Code screen when attempting to register an existing Signal account on a new device (Pixel 7a running GrapheneOS, without Google Services Framework or Play Services). 

### Steps to reproduce
- clean install of Signal APK directly from Signal website (obtained here: https://signal.org/android/apk/)
- click Continue and accept permissions
- enter the phone number of my existing Signal account
- acknowledge the prompt about Missing Google Play Services (twice)
- go through "verify you are human" captcha
- confirm the phone number again
- Verification code is sent
- Error popup shows "Unable to connect to service. Please check network connection and try again."
- I receive and paste the Verification Code
- spinner spins for 3 sec, then stops
- I remain on the Verification Code screen 

I do not have a VPN running
I am using Signal 7.11.3
Android 14
GrapheneOS, no GSF or Play services
Pixel 7a

**Actual result:** 
I am stuck on this Verification Code screen. The spinner silently disappears and I'm unable to advance. I can retry and receive another code, but it never lets me move pass this.
**Expected result:** 
I expect it to accept the code and let me into the Signal app

### Screenshots
![IMG_20240719_121923](https://github.com/user-attachments/assets/c5c7320f-8c04-49b3-8998-c751ef167954)
![Screenshot_20240719-225013](https://github.com/user-attachments/assets/42cc6a66-eff6-4344-8055-195f40baf753)

### Device info
**Device:** Pixel 7a
**Android version:** 14, GrapheneOS without GSF or Play services
**Signal version:** 7.11.3

### Link to debug log
[Signal log c8d3d5463f7f.txt](https://github.com/user-attachments/files/16318614/Signal.log.c8d3d5463f7f.txt)

**Comment by hhaslam11** [comment]
Same issue here on a brand new, stock Pixel 8

**Comment by na9y** [comment]
I get the same issue too on GrapheneOS, with the APK from Signal (https://signal.org/android/apk/).

The issue seems to be fixed in the testing version 7.12:
https://github.com/signalapp/Signal-Android/issues/13623

**Comment by ErikSkinnari** [comment]
I just went *from* graphene OS back to stock on my Pixel 8 Pro and are experiencing the exact same issue!! Have been trying all day to get the validation to work. So "glad" that it's not just me having this issue!

From the the log:
```[7.11.3] [main ] 2024-07-20 22:55:28.478 GMT+02:00 V RegistrationViewModel: handleSessionStateResult()
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: Encountered unknown error!
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: org.whispersystems.signalservice.api.push.exceptions.HttpConflictException: StatusCode: 409
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket$RegistrationCodeRequestResponseHandler.handle(PushServiceSocket.java:3066)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:2219)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket.requestVerificationCode(PushServiceSocket.java:419)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.requestSmsVerificationCode$lambda$3(RegistrationApi.kt:69)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.$r8$lambda$J6AJHpoMZvPa4GtlGHrGnI_pJ6Q(RegistrationApi.kt:0)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi$$ExternalSyntheticLambda4.fetch(R8$$SyntheticClass:0)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.api.NetworkResult$Companion.fromFetch(NetworkResult.kt:38)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.requestSmsVerificationCode(RegistrationApi.kt:68)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at org.thoughtcrime.securesms.registration.data.RegistrationRepository$requestSmsCode$2.invokeSuspend(RegistrationRepository.kt:348)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)
[7.11.3] [main ] 2024-07-20 22:55:28.479 GMT+02:00 W RegistrationViewModel: 
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W EnterCodeFragment: Binding not valid, aborting! Result: o...@...
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: Cause:
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: org.whispersystems.signalservice.api.push.exceptions.HttpConflictException: StatusCode: 409
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket$RegistrationCodeRequestResponseHandler.handle(PushServiceSocket.java:3066)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:2219)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket.requestVerificationCode(PushServiceSocket.java:419)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.requestSmsVerificationCode$lambda$3(RegistrationApi.kt:69)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.$r8$lambda$J6AJHpoMZvPa4GtlGHrGnI_pJ6Q(RegistrationApi.kt:0)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.api.registration.RegistrationApi$$ExternalSyntheticLambda4.fetch(R8$$SyntheticClass:0)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.api.NetworkResult$Companion.fromFetch(NetworkResult.kt:38)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.requestSmsVerificationCode(RegistrationApi.kt:68)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at org.thoughtcrime.securesms.registration.data.RegistrationRepository$requestSmsCode$2.invokeSuspend(RegistrationRepository.kt:348)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
[7.11.3] [main ] 2024-07-20 22:55:28.482 GMT+02:00 W UnknownError: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)```

**Comment by ErikSkinnari** [comment]
Just applied to the Beta Program and now after an update i got the message "You've made too many attempts to register this number. Please try again in 4h...." Annoying, but at least something different and good hope that it will work tomorrow.

**Comment by hhaslam11** [comment]
Joining the beta program through the Play Store fixed it on my Pixel 8. Had to wait a few hours because of the "too many attempts" message though

**Comment by Yolakalemowa** [comment]
> Joining the beta program through the Play Store fixed it on my Pixel 8. Had to wait a few hours because of the "too many attempts" message though

Hi, facing the same issue. 
Do you recommend I wait for stable release which fixes this (hopefully?) or go with beta version? I heard beta is usually discouraged for noobs like myself XD

**Comment by Yolakalemowa** [comment]
> Joining the beta program doesn't solve the issue for those of us who don't use GSF/Play services on Graphene. We rely on the standalone APK published by Signal that is specifically designed to work without those services.

Understood... so my only hope is to wait for the APK package to be updated to 7.12? 
Do you have any idea how long it usually takes to follow the github version which seems to be stable 7.12?

**Comment by jklawiter** [comment]
> > Joining the beta program doesn't solve the issue for those of us who don't use GSF/Play services on Graphene. We rely on the standalone APK published by Signal that is specifically designed to work without those services.
> 
> Understood... so my only hope is to wait for the APK package to be updated to 7.12? Do you have any idea how long it usually takes to follow the github version which seems to be stable 7.12?

You can swich to beta in gPlay and install it now.

**Comment by GHM3434** [comment]
Same issue here I can't login for over a day! Verification code doesn't get accepted. It does the loading wheel then it doesn't do anything.

**Comment by Yolakalemowa** [comment]
> > > Joining the beta program doesn't solve the issue for those of us who don't use GSF/Play services on Graphene. We rely on the standalone APK published by Signal that is specifically designed to work without those services.
> > 
> > 
> > Understood... so my only hope is to wait for the APK package to be updated to 7.12? Do you have any idea how long it usually takes to follow the github version which seems to be stable 7.12?
> 
> You can swich to beta in gPlay and install it now.

no no I don't want through google.

**Comment by Yolakalemowa** [comment]
> Same issue here I can't login for over a day! I'm able to receive the SMS code but the verification code doesn't get accepted by Signal. It does the loading wheel then it doesn't do anything.
> 
> I am trying with sandboxed google play services.

I tried that too and it didn't work. Please report back

**Comment by Yolakalemowa** [comment]
> I get the same issue too on GrapheneOS, with the APK from Signal (https://signal.org/android/apk/).
> 
> The issue seems to be fixed in the testing version 7.12: #13623

any idea when might this stable version release? this is quite ridiculous. I'm locked out of my personal/professional life until I can get into signal again.

**Comment by JAA94** [comment]
I've aplied to be a beta tester, waited one hour and then downloaded the testing version 7.12 and it all went fine with the log in. Guess that is your best shot. It's fairly simple and it worked. 




> > I get the same issue too on GrapheneOS, with the APK from Signal (https://signal.org/android/apk/).
> > The issue seems to be fixed in the testing version 7.12: #13623
> 
> any idea when might this stable version release? this is quite ridiculous. I'm locked out of my personal/professional life until I can get into signal again.

**Comment by Yolakalemowa** [comment]
> I've aplied to be a beta tester, waited one hour and then downloaded the testing version 7.12 and it all went fine with the log in. Guess that is your best shot. It's fairly simple and it worked.
> 
> > > I get the same issue too on GrapheneOS, with the APK from Signal (https://signal.org/android/apk/).
> > > The issue seems to be fixed in the testing version 7.12: #13623
> > 
> > 
> > any idea when might this stable version release? this is quite ridiculous. I'm locked out of my personal/professional life until I can get into signal again.

I never tried beta versions I heard for my noob level I should stay away from them lol. 
But maybe I can do this at least until it beocmes stable then I can jump to the stable one? (is that easy to do you think?)

**Comment by JAA94** [comment]
> > I've aplied to be a beta tester, waited one hour and then downloaded the testing version 7.12 and it all went fine with the log in. Guess that is your best shot. It's fairly simple and it worked.
> > > > I get the same issue too on GrapheneOS, with the APK from Signal (https://signal.org/android/apk/).
> > > > The issue seems to be fixed in the testing version 7.12: #13623
> > > 
> > > 
> > > any idea when might this stable version release? this is quite ridiculous. I'm locked out of my personal/professional life until I can get into signal again.
> 
> I never tried beta versions I heard for my noob level I should stay away from them lol. But maybe I can do this at least until it beocmes stable then I can jump to the stable one? (is that easy to do you think?)

It's pretty easy and it is all done through Play Store so you don't need to worry about shady procedures. Signal explain it very well in this link (https://support.signal.org/hc/en-us/articles/360007318471-Signal-Beta) 

For me it worked like a charm.

**Comment by Yolakalemowa** [comment]
> > > I've aplied to be a beta tester, waited one hour and then downloaded the testing version 7.12 and it all went fine with the log in. Guess that is your best shot. It's fairly simple and it worked.
> > > > > I get the same issue too on GrapheneOS, with the APK from Signal (https://signal.org/android/apk/).
> > > > > The issue seems to be fixed in the testing version 7.12: #13623
> > > > 
> > > > 
> > > > any idea when might this stable version release? this is quite ridiculous. I'm locked out of my personal/professional life until I can get into signal again.
> > 
> > 
> > I never tried beta versions I heard for my noob level I should stay away from them lol. But maybe I can do this at least until it beocmes stable then I can jump to the stable one? (is that easy to do you think?)
> 
> It's pretty easy and it is all done through Play Store so you don't need to worry about shady procedures. Signal explain it very well in this link (https://support.signal.org/hc/en-us/articles/360007318471-Signal-Beta)
> 
> For me it worked like a charm.

thanks but my question was about switching back to the stable one when it's released. Is that easy? or requires I reinstall signal again like a new user?

**Comment by JAA94** [comment]
> thanks but my question was about switching back to the stable one when it's released. Is that easy? or requires I reinstall signal again like a new user?

I've never rolled back to a stable version before but as far as i'm concerned and if the Signal website is correct it is as simple as this!

![imagem](https://github.com/user-attachments/assets/206590bb-94d1-4eb8-b07e-69a8680ecb4c)

**Comment by Yolakalemowa** [comment]
so this seems to be either a problem with Pixels only or with Graphene Pixels only. Because I didn't have any issues on another android phone (Tecno)...

**Comment by GHM3434** [comment]
> > Same issue here I can't login for over a day! I'm able to receive the SMS code but the verification code doesn't get accepted by Signal. It does the loading wheel then it doesn't do anything.
> > I am trying with sandboxed google play services.
> 
> I tried that too and it didn't work. Please report back

I doesn't work. It's clearly a bug with this version. I downloaded the new version from aurora store and it worked.

**Comment by Yolakalemowa** [comment]
> > > Same issue here I can't login for over a day! I'm able to receive the SMS code but the verification code doesn't get accepted by Signal. It does the loading wheel then it doesn't do anything.
> > > I am trying with sandboxed google play services.
> > 
> > 
> > I tried that too and it didn't work. Please report back
> 
> It doesn't work. It's clearly a bug with this version. I downloaded the new version from aurora store (google play store) 7.12 without joining any beta program and it worked.

wait... you sure the 7.12 version has been released publicly? I can still only see 7.11.3

**Comment by greyson-signal** [comment]
7.12 is in beta. We're taking the commit that fixed the bug and putting in a 7.11.4 prod hotfix, which has been submitted to the play store.

**Comment by Yolakalemowa** [comment]
> 7.12 is in beta. We're taking the commit that fixed the bug and putting in a 7.11.4 prod hotfix, which has been submitted to the play store.

thank you. do you know what's the usual waiting time from submitting to approval to publish on playstore?

**Comment by greyson-signal** [comment]
@Yolakalemowa The Play Console tells me it's live now :) YMMV

**Comment by jklawiter** [comment]
> > 7.12 is in beta. We're taking the commit that fixed the bug and putting in a 7.11.4 prod hotfix, which has been submitted to the play store.
> 
> thank you. do you know what's the usual waiting time from submitting to approval to publish on playstore?

It is there right now. And If you don't want to use playstore you can get it from signal webpage https://signal.org/android/apk/ 

And yes IMO the beta version should be available there also, because why not? :>
